torch_geometric/experimental.py

```diff
@@ -0,0 +1,51 @@
+from typing import Dict
+
+__experimental_options__ = {'pytorch_scatter': False}
+
+
+def is_experimental_option_enabled(option: str) -> bool:
+    r"""Returns :obj:`True`, if the experimental :attr:`option` is enabled.
+    See :class:`~torch_geometric.experimental.set_experimental_options` for
+    a list of available options."""
+    return __experimental_options__.get(option, False)
+
+
+def disable_all_experimental_options() -> None:
+    r"""Disables all experimental options."""
+    for key in __experimental_options__:
+        __experimental_options__[key] = False
+
+
+class set_experimental_options:
```

Comment:
Just a thought (if we want to avoid breaking naming conventions. We can have a class 
```
class Options:
     # does what your context manager does
```
and a function that wraps that so it can be used as a context manager
```
def option_context
     ....
     return Options(...)
```
This is how some other libraries deal with this, for example https://github.com/mwaskom/seaborn/blob/master/seaborn/rcmod.py

Reply:
Hi @Padarn, thanks for your suggestion. I don't think we break any convention here, PEP8 says this about class naming ([source](https://peps.python.org/pep-0008/#class-names)): `The naming convention for functions may be used instead in cases where the interface is documented and used primarily as a callable.` Approach taken here is also aligned with existing in [debug.py](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/debug.py).

Reply:
🆗